model-optimizer/extensions/back/ChangeOutputTypeAttributes.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+
+import numpy as np
+
+from mo.back.replacement import BackReplacementPattern
+from mo.graph.graph import Graph
+from mo.graph.graph import Node
+from mo.middle.passes.convert_data_type import data_type_str_to_np
+from mo.utils.error import Error
+
+operations_with_data_type_attributes = {
+    'Cast': {'attr_name': 'dst_type', 'in_ports_to_check': (0,)},
+    'Range': {'attr_name': 'output_type', 'in_ports_to_check': (0, 1, 2)},
+}
+
+
+class ChangeOutputTypeAttributes(BackReplacementPattern):
+    """
+    For operations that allow specify output type by attribute this transformation
+    changes output type from fp64 to fp32 since not all plugins support fp64 data type.
```

Comment:
Wording: 
"The transformation changes output type for the specific operations defined in the `operations_with_data_type_attributes` dictionary if one of the following conditions is met:
1. The operation output type is fp64. Since not all plugins support fp64 data type it is converted to fp32.
2. Keep operation output type equal to fp32 for operations located in the shape calculation sub-graphs to avoid floating point overflow.
3. Changes output type from fp32 to fp16 (and ensure that this is possible) when generating fp16 IR.
"

Reply:
done